CRM/Sqltasks/Action/APICall.php

```diff
@@ -160,15 +160,21 @@ public function execute() {
     $this->resetHasExecuted();
     $entity = $this->getConfigValue('entity');
     $action = $this->getConfigValue('action');
+    $store_api_results = $this->getConfigValue('store_api_results');
+    $data_table = $this->getDataTable();
     $parameter_specs = $this->getParameters();
 
+    if ($store_api_results) {
+      CRM_Core_DAO::executeQuery("ALTER TABLE `$data_table` ADD `id` INT AUTO_INCREMENT PRIMARY KEY");
```

Comment:
Let's check with the others in today's call if they're fine with the column name conventions we've chosen here, i.e.: with "Store (Activity ID|API Results)", your data table must not (already) have: an `id` column, a `api_result` column, an `activity_id` column and no existing `AUTO_INCREMENT` column (MySQL only allows one per table).